### PR TITLE
Add rangeless/cci.20201117

### PR DESCRIPTION
**Pull request by hnOsmium0001**
Specify library name and version:  **rangeless/cci.20201117**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/rangeless/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from conans import ConanFile, tools
+import os
+import glob
+
+class RangelessConan(ConanFile):
+    name = "rangeless"
+    description = "c++ LINQ -like library of higher-order functions for data manipulation"
+    license = "Public domain"
```
Comment (prince-chrismc) [inline recipes/rangeless/all/conanfile.py:8]: 🤯 

https://github.com/ast-al/rangeless/blob/master/LICENSE

https://www.ncbi.nlm.nih.gov/home/about/policies/

😕

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/rangeless/all/conanfile.py (hunk 1 above)
Comment (hnOsmium0001) [reply recipes/rangeless/all/conanfile.py:8]: (Not a lawyer)
Now I read it the second time, I think this is basically MIT (public domain redistribution rights + requirements to cite the source). The author just happen to work for NCBI, so the organization name is mentioned.
Though that means it's not "Public domain" anymore. What would be the appropriate name to put here?

Comment (prince-chrismc) [reply recipes/rangeless/all/conanfile.py:8]: I haven't a clue what we should do...

```suggestion
    license = "rangless-public-domain"
```

@uilianries when you have a change, what should the license be?

Comment (uilianries) [reply recipes/rangeless/all/conanfile.py:8]: I agree to MIT.

CC0 anyone can re-license on any terms, and license copy is not mandatory. As that license requires citation about NCBI, it fits better for MIT.

https://choosealicense.com/licenses/cc0-1.0/
https://choosealicense.com/licenses/mit/

Comment (prince-chrismc) [reply recipes/rangeless/all/conanfile.py:8]: Thank you for your input!

Comment (uilianries) [reply recipes/rangeless/all/conanfile.py:8]: I would love to know what NCBI thinks about, I'll send an email asking about and updating here when having an answer.

Comment (uilianries) [reply recipes/rangeless/all/conanfile.py:8]: Well, I didn't find any email address, only center support which is a FAQ actually.

**Review by prince-chrismc — APPROVED** [review]

File: recipes/rangeless/all/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "cci.20201117":
```
Comment (mathbunnyru) [inline recipes/rangeless/all/conandata.yml:2]: Could you please create an issue asking the upstream project to release versions from time to time?

File: recipes/rangeless/all/test_package/test_package.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#include <vector>
+#include <utility>
+#include <fn.hpp>
+
+namespace fn = rangeless::fn;
+using fn::operators::operator%;   // arg % fn   equivalent to fn(std::forward<Arg>(arg))
+using fn::operators::operator%=;  // arg %= fn; equivalent to arg = fn( std::move(arg));
```
Comment (mathbunnyru) [inline recipes/rangeless/all/test_package/test_package.cpp:7]: This looks awesome :)

**Review by mathbunnyru — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/rangeless/all/conandata.yml (hunk 2 above)
Comment (hnOsmium0001) [reply recipes/rangeless/all/conandata.yml:2]: I would rather not for 2 reasons:
1. CCI allows commit snapshot based versioning perfectly fine, no need to complicate this for now.
2. It seems like the author hasn't been maintaining that project. It won't add too much value (you are probably **only** going to get a 1.0.0) to properly version it.

**Pull request merged**